utils/catalog_production/add_glsn_properties.py

```diff
@@ -0,0 +1,286 @@
+    for sed_bin in sed_label:
+        columns.append(sed_bin)
+    data = catalog.get_quantities(columns,
+                                  filters=['mag_true_i_lsst > %f' % i_gal_bright,
+                                           'mag_true_i_lsst < %f' % i_gal_dim,
+                                           'redshift_true > %f' % z_cat_min,
+                                           'redshift_true < %f' % z_cat_max,
+                                           'stellar_mass_bulge/stellar_mass > 0.99'])
+    data_df = pd.DataFrame(data)
+    data_df.to_csv('sn_matching_checkpoint_2.csv', index=False)
+
+
+def get_30_band_mags(gcr_glsn_match, data_df, catalog):
+
+    H0 = catalog.cosmology.H0.value
+    Om0 = catalog.cosmology.Om0
+
+    sed_name_list = []
+    magNorm_list = []
+    i = 0
+    lsst_mags = []
+    mag_30_list = []
+    redshift_list = []
+
+    sed_label = []
+    sed_min_wave = []
+    sed_wave_width = []
+    for quant_label in sorted(catalog.list_all_quantities()):
+        if (quant_label.startswith('sed') and quant_label.endswith('bulge')):
+            sed_label.append(quant_label)
+            label_split = quant_label.split('_')
+            sed_min_wave.append(int(label_split[1])/10)
+            sed_wave_width.append(int(label_split[2])/10)
+    bin_order = np.argsort(sed_min_wave)
+    sed_label = np.array(sed_label)[bin_order]
+    sed_min_wave = np.array(sed_min_wave)[bin_order]
+    sed_wave_width = np.array(sed_wave_width)[bin_order]
+
+    for gal_id, gal_z in zip(gcr_glsn_match[:, 1], 
+                             gcr_glsn_match[:, 0]):
+        
+        if i % 100 == 0:
+            print(i)
+        i+=1
+        #print(np.int(gal_id), type(data_df['galaxy_id'].iloc[0]))
+        #print(gal_z, type(gal_z), data_df['redshift_true'].iloc[0])
+        #print(int(gal_id))
+        data_subset = data_df.query(str('galaxy_id == %i' % np.int(gal_id)))# + ' and ' + 'redshift_true == {}'.format(gal_z))) ## Galaxy Ids are not unique in cosmoDC2_v0.1
+        #print(data_subset)
+        mag_array = []
+        lsst_mag_array = [data_subset['mag_%s_lsst' % band_name].values[0] for band_name in ['u', 'g', 'r', 'i', 'z', 'y']]
+        for sed_bin in sed_label:
+            mag_array.append(-2.5*np.log10(data_subset[sed_bin].values[0]))
+        mag_array = np.array(mag_array)
+        lsst_mag_array = np.array(lsst_mag_array)
+        lsst_mags.append(lsst_mag_array)
+        mag_30_list.append(mag_array)
+        redshift_list.append(gal_z)
+    
+    mag_30_list = np.array(mag_30_list).T
+    lsst_mags = np.array(lsst_mags).T
+    redshift_list = np.array(redshift_list)
+
+    print('Getting SEDs')
+
+    sed_name, magNorm, av, rv = sed_from_galacticus_mags(mag_30_list, redshift_list,
+                                                         H0, Om0, sed_min_wave, sed_wave_width, lsst_mags)
+
+    return sed_name, magNorm, av, rv
+
+
+if __name__ == "__main__":
+
+    catalog_version = sys.argv[1]
```

Comment:
Please use `argparse.ArgumentParser()` and include a `help` entry for each argument so that someone who is not us can figure out how to run these scripts.